### PR TITLE
rfc003 changes due to rfc011/rfc012

### LINKED ISSUE #20: Expand RFC003 when VC for "name" is available
§5.2.1.5 of the OAuth flow describes that the login contract must be matched to the issuer. This is done by name/place in the login contract. The name/place VC is required to do the matching.

### PR DESCRIPTION
**Pull request by woutslakhorst**
Closes #20 

create 2 issues after merge:
- set correct `aud` in bearer token nuts-foundation/nuts-node#83
- do correct contract validation with NutsOrganizationCredential

### CODE REVIEW

File: rfc/rfc003-oauth2-authorization.md [hunk 1]
```diff
@@ -220,11 +220,13 @@ The JWT **iat** and **exp** fields MUST be validated. The timestamp of validatio
...
+The **usi** field in the JWT contains the signed login contract. If present it MUST validate according to the [Authentication Token RFC](rfc002-authentication-token.md). The login contract MUST contain the `name` and `city` of the actor.
+The `name` and `city` MUST match with a valid [Verifiable Credential](rfc011-verifiable-credential.md) issued to the actor. This must be a [NutsOrganizationCredential](rfc012-nuts-organization-credential.md).
 
 **5.2.1.6. Endpoint validation**
 
-The **aud** field MUST match the registered endpoint from which the authorization server is listening. This prevents the use of the JWT at any other endpoint. The endpoint reference is used for this. \([RFC7523](https://tools.ietf.org/html/rfc7523#section-3)\)
+The **aud** field MUST match the identifier of the registered endpoint. This prevents the use of the JWT at any other endpoint. The endpoint reference is used for this. \([RFC7523](https://tools.ietf.org/html/rfc7523#section-3)\).
```
Comment (stevenvegt) [inline rfc/rfc003-oauth2-authorization.md:228]: Perhaps emphasize that it must be the identifier of the the authorization server, and not that of the bolt service endpoint where the token will be used.

File: rfc/rfc003-oauth2-authorization.md [hunk 2]
```diff
@@ -112,7 +112,7 @@ In order for the client to resolve the authorization server endpoint it MUST loo
 * **iss**: The issuer MUST contain the DID of the actor, thus the care organization making the request.
 * **sub**: The subject MUST contain the DID of the custodian. The custodian's DID could be used to find the relevant consent \(together with the actor and subject\).
 * **sid**: The Nuts subject id, patient identifier in the form of an oid encoded BSN. Optional.
-* **aud**: As per [RFC7523](https://tools.ietf.org/html/rfc7523), the aud MUST be the DID listed under the `auth` key of the services serviceEndpoint.
+* **aud**: As per [RFC7523](https://tools.ietf.org/html/rfc7523), the aud MUST be the service identifier from the DID Document. The service MUST be of type: `oauth`.
```
Comment (reinkrul) [inline rfc/rfc003-oauth2-authorization.md]: which's party DID document?

File: rfc/rfc003-oauth2-authorization.md [hunk 3]
```diff
@@ -96,8 +96,8 @@ Example of the actor's DID document:
 
 #### 4.1.2 Server registration
 
-Each service MUST define an `oauth` serviceEndpoint. This endpoint is either the URL of the authorization server, or in case of a compound service an URI which resolves to another service.
-In order for the client to resolve the authorization server endpoint it MUST look up the `auth` service endpoint of the service.
+Each service MUST define an `oauth` serviceEndpoint. This endpoint refers to another service in the DID Document with the type: `oauth`.
```
Comment (reinkrul) [inline rfc/rfc003-oauth2-authorization.md:99]: > Each service

"Service" is ambigious: OAuth service ("server") or DID service?

> refers to another service in the DID document

the same DID document, or a parent's?

File: rfc/rfc003-oauth2-authorization.md (hunk 2 above)
Comment (reinkrul) [inline rfc/rfc003-oauth2-authorization.md]: MUST be an absolute endpoint (in other words, not a compound one)

File: rfc/rfc003-oauth2-authorization.md [hunk 4]
```diff
@@ -220,11 +220,13 @@ The JWT **iat** and **exp** fields MUST be validated. The timestamp of validatio
 
 **5.2.1.5. Login contract validation**
 
-The **usi** field in the JWT contains the signed login contract. If present it MUST validate according to the [Authentication Token RFC](rfc002-authentication-token.md). The login contract MUST contain the name of the actor.
+The **usi** field in the JWT contains the signed login contract. If present it MUST validate according to the [Authentication Token RFC](rfc002-authentication-token.md). The login contract MUST contain the `name` and `city` of the actor.
+The `name` and `city` MUST match with a valid [Verifiable Credential](rfc011-verifiable-credential.md) issued to the actor. This must be a [NutsOrganizationCredential](rfc012-nuts-organization-credential.md).
```
Comment (reinkrul) [inline rfc/rfc003-oauth2-authorization.md:224]: Mention that name and city match must be an exact, case sensitive match

**Review by reinkrul — CHANGES_REQUESTED** [review]

File: rfc/rfc003-oauth2-authorization.md (hunk 3 above)
Comment (woutslakhorst) [reply rfc/rfc003-oauth2-authorization.md:99]: Added notion of compound service (also in terminology)

**Review by reinkrul — APPROVED** [review]

**Pull request merged**